include/boost/geometry/formulas/andoyer_inverse.hpp

```diff
@@ -73,28 +74,44 @@ class andoyer_inverse
 
         CT const c0 = CT(0);
         CT const c1 = CT(1);
-        CT const pi = math::pi<CT>();
         CT const f = formula::flattening<CT>(spheroid);
 
-        CT const dlon = lon2 - lon1;
-        CT const sin_dlon = sin(dlon);
-        CT const cos_dlon = cos(dlon);
+        CT d; //the spherical distance
+        CT dlon;
+        CT cos_d;
+        CT cos_dlon;
+        CT sin_dlon;
+        CT cos_lat1;
+        CT cos_lat2;
```

Comment:
These variables are not initialized which is wrong by itself but even worse later thay're used for azimuth and quantities calculation. Something meaningful is assigned to them only if `! math::equals(lon1, lon2)` so uninitialized variables may be used below. Or am I missing something?

In general I think it is desireable to define and initialize variabels at once, if that'd require to redesign the algorithm/formula then maybe this would be the way to go?